src/arch-syscall-check.c

```diff
@@ -90,6 +94,15 @@ int main(int argc, char *argv[])
 		}
 		sys_name = sys->name;
 
+		syscall_num = syscall_resolve_name(sys_name, OFFSET_ARCH(x86_64));
+		if (syscall_num >= 0) {
+			sys_name_tmp = syscall_resolve_num(syscall_num, OFFSET_ARCH(x86_64));
+			if (strcmp (sys_name, sys_name_tmp)) {
+				printf("syscall_resolve returns wrong data\n");
+				return 1;
+			}
+		}
+
```

Comment:
You lost me here, what is the purpose of this addition?  Is it just to make sure `syscall_resolve_name()` and `syscall_resolve_num()` are working?

Reply:
the lookup code is much bigger now, without such check that verifies every single syscall the code coverage will drop significantly.  The main reason for having it is code coverage, and since the lookup now is way more complex (at least for the generated code), it makes sense to test it for each syscall.

Reply:
If this is purely for a testing, it should be part of a test and not in `arch-syscall-check.c`.  You could create a new test, maybe similar to 15-basic-resolver, that would exercise the syscall lookup APIs.  As you say, the code is much more complex now, a new test to exercise it would be a good idea.